File: Lab-4-lambda/node_10.18.1/create_report_code.js
exports.handler = async function(event, context){
  var 
      decoded = {},
      return_me = {
          "message_str": "Report processing, check your phone shortly"
      };
  if(event.params && event.params.header && event.params.header.Authorization){
    decoded = JSON.parse(new Buffer(event.params.header.Authorization.replace("Bearer ", "").trim().split(".")[1], "base64").toString("binary"));
    console.log(decoded);
    return_me.cell_str = decoded.phone_number;
    return_me.name_str = decoded["cognito:username"];
    return_me.ipv4_str = event.params.header["X-Forwarded-For"];
    return_me.message_str = "Report Processing";
  }else{
    console.log("testing in the console without AUTH");
  }
  return return_me;
};

File: Lab-4-lambda/python_3.6.8/create_report_code.py
import json
import base64

def lambda_handler(event, context):
    token_str = ""
    ipv4_str = ""
    decoded = {}
    return_me = {}
    name_str = ""
    extract_str = ""
    cell_str = ""
    return_me["message_str"] =  "Report processing, check your phone shortly"
    if "params" in event:
        #i.e we are at the website and have a valid Bearer token passed
        token_str = event["params"]["header"]["Authorization"]
        extract_str = token_str.replace("Bearer ", "").strip().split(".")[1] #.toString("binary"));
        decoded = json.loads(base64.b64decode(extract_str))
        cell_str = decoded["phone_number"]
        name_str = decoded["cognito:username"]
        ipv4_str = event["params"]["header"]["X-Forwarded-For"]
        return_me["cell_str"] = cell_str
        return_me["name_str"] = name_str
        return_me["ipv4_str"] = ipv4_str
        return_me["message_str"] = "Report Processing"
    return return_me
